File: app/src/types/booking.ts
export type BookingSource = "WalkIn" | "Appointment";
export type BookingStatus = "Open" | "Completed" | "Cancelled";

export interface Booking {
  id: number;
  source: BookingSource;
  appointmentId?: number | null;
  customerName?: string | null;
  phoneNumber?: string | null;
  serviceOfferingId: number;
  serviceName: string;
  servicePrice?: number | null;
  bookedAt: string;
  status: BookingStatus;
  notes?: string | null;
  createdAt: string;
  lastModifiedAt?: string | null;
}

export interface SaveBookingRequest {
  source: BookingSource;
  appointmentId?: number | null;
  customerName?: string | null;
  phoneNumber?: string | null;
  serviceOfferingId: number;
  bookedAt: string;
  status: BookingStatus;
  notes?: string | null;
}

File: backend/src/Sixram.Nextgen.JinmarxWellness.Domain/Entities/Booking.cs
using Sixram.Nextgen.JinmarxWellness.Domain.Common;

namespace Sixram.Nextgen.JinmarxWellness.Domain.Entities;

public class Booking : AuditableEntity
{
    public BookingSource Source { get; set; } = BookingSource.WalkIn;
    public int? AppointmentId { get; set; }
    public Appointment? Appointment { get; set; }
    public string? CustomerName { get; set; }
    public string? PhoneNumber { get; set; }
    public int ServiceOfferingId { get; set; }
    public ServiceOffering? ServiceOffering { get; set; }
    public string ServiceName { get; set; } = string.Empty;
    public decimal? ServicePrice { get; set; }
    public DateTimeOffset BookedAt { get; set; }
    public BookingStatus Status { get; set; } = BookingStatus.Open;
    public string? Notes { get; set; }
}

public enum BookingSource
{
    WalkIn = 0,
    Appointment = 1
}

public enum BookingStatus
{
    Open = 0,
    Completed = 1,
    Cancelled = 2
}
